### ISSUE TITLE
Package summary not updated for newer builds

### ISSUE THREAD

**Issue by mbargull** [Anaconda-Platform/anaconda-client#477]
ref: https://github.com/conda-forge/freecad-feedstock/issues/7

The summary on https://anaconda.org/conda-forge/freecad does not reflect the changes added in a later build: https://github.com/conda-forge/freecad-feedstock/pull/2/files#diff-e178b687b10a71a3348107ae3154e44cR79

```sh
$ anaconda --version && anaconda search -t conda freecad 2>&1 | grep -A1 conda-forge                                                                                               
anaconda Command line client (version 1.6.14)
     conda-forge/freecad       |   0.18b0 | conda           | linux-64, win-64, osx-64 | py36h0c05c59_2, py36h1ae927f_2, py36h275e31e_2, py36h0c05c59_0, py36h0c05c59_1
                                          : Simple, fast, extensible JSON encoder/decoder for Python
```

(PS: Should I rather raise the issue over at https://github.com/continuumio/anaconda-issues/issues? I'm not sure which part is to blame, so I chose the less crowded issue tracker...)

**Comment by goanpeca** [comment]
@mbargull will work on this in the coming weeks. 

Could you help me narrow this issue a bit?

If one uploads a new (conda) package `pysomething` it has the correct summary but if 
we add newer versions, the summary is not updated, if it is now something different on the newer release? 

If this is the case I know why it is happening, but it would probably require making some changes on Anaconda Cloud as well, since I believe prior to uploading packages, a "Package" is created and is during this process that the summary is defined, hence why updating with newer versions does not update the summary. The changes on Anaconda Cloud and Anaconda Client side are probably small enough.

**Comment by mbargull** [comment]
Yes, if a newer build with a different summary/description is uploaded, these changes are not reflected on anaconda.org (either through the client or web interface).
I think what I would expect is that the displayed summary/description always corresponds to the latest build of the latest version of a package. E.g., with the following upload sequence:
```
pkg[version=1.0, build_number=0] with summary "Good package!"
pkg[version=2.0, build_number=0] with summary "Great package!"
pkg[version=2.0, build_number=1] with summary "Awesome package!"
pkg[version=1.0, build_number=1] with summary "Good package!"
```
I would expect `"Awesome package!"` to be displayed since that build has the highest version and build number.
If there are diverging version/build numbers in different subdirs, I would just choose the highest version/build of the union of the subdirs.
If we have different labels, I would only look at the `main` label if that one is present and disregard any others (even if those have a higher version package available, e.g., prereleases). If we don't have a build under the `main` label, I would just take the union of all builds under all labels.
Also, if a build is removed or moved to another label, I would expect the summary/description to be updated accordingly.

**Comment by goanpeca** [comment]
Thanks for the input @mbargull, all those changes need to be handled mostly at the Anaconda Cloud level, but we can keep it here to track it.

**Comment by bkreider** [comment]
I think the issue here is the way these things are abstracted at the Repo level.  I'm fuzzy on this and perhaps the code has changed, but I think these are the objects involved.  These get created behind the scenes the first time a new package is uploaded:

1.  Packages:  Name, description, license, public/private, etc
  1.1 Release 1.0:  Version of a package
     1.1.1 distribution/file:  The conda package build - Package name-version-arch-build number:  an actual conda package
  1.2 Release 1.1: Version of a package
  1.3 Release 2.0: Version of a package

The description is stored at the global level of the Package object.  If you change the description it changes for all "builds".   

I would assume that the --description argument only gets used by anaconda-client, if it needs to create a "Package" object on the very first upload of a package.

**Comment by luzpaz** [comment]
Any progress on this?
